backend/infrahub/core/query/node.py

```diff
@@ -894,18 +910,12 @@ async def query_init(self, db: InfrahubDatabase, **kwargs: Any) -> None:
             """ % {"branch_filter": branch_filter, "node_kind": self.schema.kind}
             self.add_to_query(topquery)
 
-        use_simple = False
         if self.has_filter_by_id and self.filters:
-            use_simple = True
-            where_clause_elements.append("n.uuid = $uuid")
             self.params["uuid"] = self.filters["id"]
-        elif not self.has_filters and not self.schema.order_by:
-            use_simple = True
+            self.add_to_query(" AND n.uuid = $uuid")
+            return
+        if not self.has_filters and not self.schema.order_by:
```

Comment:
should we also check if order is disabled here and exit early ?

Reply:
If this condition is met, we set `self.order_by` to `["n.uuid"]` in order to guarantee pagination, and we exit early. As `schema.order_by` is empty, it feels like we do not even need to check for `order`?